front/lib/api/assistant/generation.ts

```diff
@@ -238,7 +238,10 @@ export async function constructPrompt(
 
   // if meta includes the string "{ASSISTANTS_LIST}"
   if (meta.includes("{ASSISTANTS_LIST}")) {
-    const agents = await getAgentConfigurations(auth);
+    const agents = await getAgentConfigurations(
+      auth,
+      auth.user() ? "list" : "public"
```

Comment:
OK I see now. v1 has no user but is auth.isUser() = true

We def want to check that otherwise I can leak information from one workspace to another. So indeed we want to support no user but we def want to check auth.isUser() :+1:

Reply:
clear as per irl